Flaky cannotLeaveWhenNewConfigurationDoesNotHaveQuorum: io.atomix.raft.ReconfigurationTest$Leaving

**Summary**

> Try to answer the following as best as possible

- How often does the test fail? - observed 2-3 failures
- Does it block your work?
- Do we suspect that it is a real failure?

**Failures**

```
java.lang.AssertionError: 

Expecting actual throwable to be an instance of:
  java.util.concurrent.ExecutionException
but was:
  java.util.concurrent.TimeoutException
	at java.base/java.util.concurrent.CompletableFuture.timedGet(CompletableFuture.java:1960)
	at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2095)
	at org.assertj.core.internal.Futures.assertFailedWithin(Futures.java:118)
	...(71 remaining lines not displayed - this can be changed with Assertions.setMaxStackTraceElementsDisplayed)
	at io.atomix.raft.ReconfigurationTest$Leaving.cannotLeaveWhenNewConfigurationDoesNotHaveQuorum(ReconfigurationTest.java:542)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at java.base/java.util.concurrent.RecursiveAction.exec(RecursiveAction.java:194)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:387)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1312)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1843)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1808)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:188)
```

**Logs**

[Test results for unit tests.zip](https://github.com/camunda/zeebe/files/14496385/Test.results.for.unit.tests.zip)

--- a/zeebe/atomix/cluster/src/test/java/io/atomix/raft/ReconfigurationTest.java
+++ b/zeebe/atomix/cluster/src/test/java/io/atomix/raft/ReconfigurationTest.java
@@ -538,7 +538,7 @@ final class ReconfigurationTest {
       assertThat(m2.leave())
           .describedAs(
               "Should fail to leave because quorum not available for the new configuration")
-          .failsWithin(Duration.ofSeconds(2))
+          .failsWithin(Duration.ofSeconds(10))
           .withThrowableOfType(ExecutionException.class);
     }
 

